File: GitAndroid/app/src/main/java/com/xiangyu/material/gitandroid/VipName.java
package com.xiangyu.material.gitandroid;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by shixiangyu on 2019-10-30.
 */
public class VipName implements Parcelable {
    public String name;
    public String price;

    public VipName() {

    }

    protected VipName(Parcel in) {
        name = in.readString();
        price = in.readString();
    }

    public static final Creator<VipName> CREATOR = new Creator<VipName>() {
        @Override
        public VipName createFromParcel(Parcel in) {
            return new VipName(in);
        }

        @Override
        public VipName[] newArray(int size) {
            return new VipName[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(price);
    }
}

File: GitAndroid/app/src/main/java/com/xiangyu/material/gitandroid/VipName.kt
package com.xiangyu.material.gitandroid

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by shixiangyu on 2019-10-30.
 */
class VipName() : Parcelable {
    var name: String? = null
    var price: String? = null

    constructor(parcel: Parcel) : this() {
        name = parcel.readString()
        price = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeString(price)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<VipName> {
        override fun createFromParcel(parcel: Parcel): VipName {
            return VipName(parcel)
        }

        override fun newArray(size: Int): Array<VipName?> {
            return arrayOfNulls(size)
        }
    }
}